builds/e2e/templates/e2e-setup.yaml

```diff
@@ -0,0 +1,140 @@
+      Write-Output '##vso[task.setvariable variable=whichImageBuild]latestFromBranch'
+      Write-Output '##vso[task.setvariable variable=packageBuildId]$(Build.TriggeredBy.BuildId)'
+    }
+    # 2) 'Build Images' build triggered this build (use latest 'Edgelet Packages' build on master)
+    elseif ('$(Build.TriggeredBy.DefinitionName)' -eq '$(azure.pipeline.images)')
+    {
+      Write-Output '##vso[task.setvariable variable=whichPackageBuild]latestFromBranch'
+      Write-Output '##vso[task.setvariable variable=whichImageBuild]specific'
+      Write-Output '##vso[task.setvariable variable=imageBuildId]$(Build.TriggeredBy.BuildId)'
+    }
+    # 3) Manual build (use given args for 'Build Images' build ID and 'Edgelet Packages' build ID)
+    elseif ('$(az.pipeline.packages.buildId)' -and '$(az.pipeline.images.buildId)')
+    {
+      Write-Output '##vso[task.setvariable variable=whichPackageBuild]specific'
+      Write-Output '##vso[task.setvariable variable=whichImageBuild]specific'
+      Write-Output '##vso[task.setvariable variable=packageBuildId]$(az.pipeline.packages.buildId)'
+      Write-Output '##vso[task.setvariable variable=imageBuildId]$(az.pipeline.images.buildId)'
+    }
+    else # Error
+    {
+      if ('$(Build.TriggeredBy.DefinitionName)' -ne '')
+      {
+        Write-Output "Build triggered by unrecognized build '$(Build.TriggeredBy.DefinitionName)'"
+      }
+      else
+      {
+        Write-Output "Missing build parameter 'az.pipeline.packages.buildId' or 'az.pipeline.images.buildId'"
+      }
+      return 1
+    }
+  displayName: Locate edgelet packages, runtime images
+
+- task: DownloadBuildArtifacts@0
+  displayName: Download edgelet packages
+  inputs:
+    buildType: specific
+    project: one
+    pipeline: $(az.pipeline.packages)
+    buildVersionToDownload: $(whichPackageBuild)
+    branchName: refs/heads/master
+    buildId: $(packageBuildId)
+    downloadType: specific
+    itemPattern: $(artifactDir)/*
+
+- task: DownloadBuildArtifacts@0
+  displayName: Get Docker image info
+  inputs:
+    buildType: specific
+    project: one
+    pipeline: $(az.pipeline.images)
+    buildVersionToDownload: $(whichImageBuild)
+    branchName: refs/heads/master
+    buildId: $(imageBuildId)
+    downloadType: specific
+    itemPattern: $(az.pipeline.images.artifacts)/artifactInfo.txt
+
+- pwsh: |
+    $certsDir = '$(System.ArtifactsDirectory)/certs'
+    New-Item "$certsDir" -ItemType Directory -Force | Out-Null
+    $env:ROOT_CERT | Out-File -Encoding Utf8 "$certsDir/rsa_root_ca.cert.pem"
+    $env:ROOT_KEY | Out-File -Encoding Utf8 "$certsDir/rsa_root_ca.key.pem"
+    Write-Output "##vso[task.setvariable variable=certsDir]$certsDir"
+  displayName: Install CA keys
+  env:
+    ROOT_CERT: $(TestRootCaCertificate)
+    ROOT_KEY: $(TestRootCaKey)
+
+- pwsh: |
+    $testDir = '$(Build.SourcesDirectory)/test/Microsoft.Azure.Devices.Edge.Test'
+    dotnet build $testDir
```

Comment:
```suggestion
    dotnet build $testDir -c Release
```

Reply:
Why? You get better callstacks in debug mode...

Reply:
I think e2e tests should be run as close as production; therefore I think release configuration should be used.  correct me if I'm wrong.

Reply:
It's just the test code that's built in debug mode. The tests get iotedged from release packages (DEB/CAB), and Edge Agent/Hub are release docker images.